core/src/components/range/range.tsx

```diff
@@ -432,10 +433,8 @@ export class Range implements ComponentInterface {
     const barStart = `${ratioLower * 100}%`;
     const barEnd = `${100 - ratioUpper * 100}%`;
 
-    const doc = document;
-    const isRTL = doc.dir === 'rtl';
-    const start = isRTL ? 'right' : 'left';
-    const end = isRTL ? 'left' : 'right';
+    const start = isRTL(this.el) ? 'right' : 'left';
+    const end = isRTL(this.el) ? 'left' : 'right';
```

Comment:
Minor optimization, but we can cache `isRTL(this.el)` to a variable so we don't need to call it twice.